activerecord/test/cases/associations/has_one_through_disable_joins_associations_test.rb

```diff
@@ -0,0 +1,416 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+require "models/club"
+require "models/member_type"
+require "models/member"
+require "models/membership"
+require "models/sponsor"
+require "models/organization"
+require "models/member_detail"
+require "models/minivan"
+require "models/dashboard"
+require "models/speedometer"
+require "models/category"
+require "models/author"
+require "models/essay"
+require "models/owner"
+require "models/post"
+require "models/comment"
+require "models/categorization"
+require "models/customer"
+require "models/carrier"
+require "models/shop_account"
+require "models/customer_carrier"
+
+class HasOneThroughDisableJoinsAssociationsTest < ActiveRecord::TestCase
+  fixtures :member_types, :members, :clubs, :memberships, :sponsors, :organizations, :minivans,
+           :dashboards, :speedometers, :authors, :author_addresses, :posts, :comments, :categories, :essays, :owners
+
+  def setup
+    @member = members(:groucho)
+  end
+
+  def test_has_one_through_with_has_one
+    assert_equal clubs(:boring_club), @member.club_without_joins
+  end
+
+  def test_has_one_through_executes_limited_query
```

Comment:
This one fails locally and I wonder if this test even makes sense when we're not joining tables.

```
Query pattern(s) /LIMIT|ROWNUM <=|FETCH FIRST/ not found.
Queries:
SELECT "memberships"."club_id" FROM "memberships" WHERE "memberships"."member_id" = ?
SELECT "clubs".* FROM "clubs" WHERE "clubs"."id" IN (?, ?, ?, ?) AND "categories"."name" = ?
```